yarn-project/simulator/src/avm/opcodes/external_calls.ts

```diff
@@ -42,20 +41,30 @@ export class Call extends Instruction {
     super();
   }
 
-  // TODO(https://github.com/AztecProtocol/aztec-packages/issues/3992): there is no concept of remaining / available gas at this moment
-  async execute(context: AvmContext): Promise<void> {
-    const [_gasOffset, addrOffset, argsOffset, retOffset, successOffset] = Addressing.fromWire(this.indirect).resolve(
-      [this._gasOffset, this.addrOffset, this.argsOffset, this.retOffset, this.successOffset],
+  async run(context: AvmContext): Promise<void> {
```

Comment:
Would it make sense to still have run as a bit of a wrapper (doing gas stuff) and then have execute do the non-gas part?

(feel free to ignore if you'd have to repeat stuff given that execute takes only a context)

Reply:
Not really, since I need to read the inputs from memory to be able to both execute and run. Maybe this hints at a wrong abstraction RE gas processing? I'll think more about it.